plugin/trino-faker/src/main/java/io/trino/plugin/faker/FakerMetadata.java

```diff
@@ -388,11 +434,234 @@ public synchronized Optional<ConnectorOutputMetadata> finishCreateTable(
+            return info;
+        }
+        Map<String, Object> minimums = new LinkedHashMap<>();
+        Map<String, Object> maximums = new HashMap<>();
+        Map<String, Long> distinctValues = new HashMap<>();
+        List<ColumnInfo> columns = info.columns();
+        Map<String, Type> types = columns.stream().collect(toMap(ColumnInfo::name, ColumnInfo::type));
+        computedStatistics.forEach(statistic -> statistic.getColumnStatistics().forEach((metadata, block) -> {
+            checkState(block.getPositionCount() == 1, "Expected a single position in aggregation result block");
+            String columnName = metadata.getColumnName();
+            Type type = types.get(columnName);
+            if (metadata.getStatisticType().equals(ColumnStatisticType.MIN_VALUE)) {
+                // TODO overwrite, or compare?
+                Object minimum = readNativeValue(type, block, 0);
+                if (minimum != null) {
+                    minimums.put(columnName, minimum);
+                }
+            }
+            else if (metadata.getStatisticType().equals(ColumnStatisticType.MAX_VALUE)) {
+                Object maximum = readNativeValue(type, block, 0);
+                if (maximum != null) {
+                    maximums.put(columnName, maximum);
+                }
+            }
+            else if (metadata.getStatisticType().equals(ColumnStatisticType.NUMBER_OF_DISTINCT_VALUES)) {
+                distinctValues.put(columnName, (long) firstNonNull(readNativeValue(BIGINT, block, 0), 0));
+            }
+            else {
+                throw new IllegalArgumentException("Unexpected column statistic type: " + metadata.getStatisticType());
+            }
+        }));
+
+        if (minimums.isEmpty() && distinctValues.isEmpty()) {
+            return info;
+        }
+
+        Map<String, List<Object>> columnValues = getColumnValues(tableName, info, distinctValues, minimums, maximums);
+
+        return info.withColumns(columns.stream().map(column -> {
+            if (isCharacterType(column.type()) || !minimums.containsKey(column.name())) {
+                return column;
+            }
+            ValueSet valueSet;
+            if (columnValues.containsKey(column.name())) {
+                valueSet = ValueSet.copyOf(column.type(), columnValues.get(column.name()));
+            }
+            else {
+                valueSet = ValueSet.ofRanges(Range.range(
+                        column.type(),
+                        minimums.get(column.name()),
+                        true,
+                        maximums.get(column.name()),
+                        true));
+            }
+            return column.withHandle(column.handle().withDomain(Domain.create(valueSet, false)));
+        }).collect(toImmutableList()));
+    }
+
+    private synchronized void createViewWithPredicates(ConnectorSession session, SchemaTableName tableName, TableInfo info, Collection<ComputedStatistics> computedStatistics)
```

Comment:
I thought about it more, and we definitely should introduce a new property for sequences, instead of using a view. It would make it easier to support different data types, and the UX would be better - creating an additional view when a user creates a table is surprising and confusing.

Reply:
https://github.com/trinodb/trino/pull/24590

Reply:
Updated!